### PR TITLE
Add UpdateServicePrimaryTaskSet to ECS deployment

### PR DESCRIPTION
**Pull request by TaKO8Ki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/cloudprovider/ecs/client.go [hunk 1]
```diff
@@ -139,6 +142,7 @@ func (c *client) CreateTaskSet(ctx context.Context, service types.Service, taskD
 		Cluster:        service.ClusterArn,
 		Service:        service.ServiceArn,
 		TaskDefinition: taskDefinition.TaskDefinitionArn,
+		Scale:          &types.Scale{Unit: types.ScaleUnitPercent, Value: 100},
```
Comment (khanhtc1202) [inline pkg/app/piped/cloudprovider/ecs/client.go:145]: should we replace this `100` value here with the real value passed by configuration 👀

File: pkg/app/piped/cloudprovider/ecs/ecs.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type Client interface {
 	DeregisterTaskDefinition(ctx context.Context, taskDefinition types.TaskDefinition) (*types.TaskDefinition, error)
 	CreateTaskSet(ctx context.Context, service types.Service, taskDefinition types.TaskDefinition) (*types.TaskSet, error)
 	DeleteTaskSet(ctx context.Context, service types.Service, taskSet types.TaskSet) (*types.TaskSet, error)
+	UpdateServicePrimaryTaskSet(ctx context.Context, service types.Service, taskSet types.TaskSet) (*types.TaskSet, error)
```
Comment (khanhtc1202) [inline pkg/app/piped/cloudprovider/ecs/ecs.go:43]: I just realized that the signature of our functions are full of `passes struct and returns pointers` pattern 😂 Should we make another PR to refactor it as `passes pointers and returns pointers` 👀 Ofcourse will do it after the sync function works perfectly as expected 😄

File: pkg/app/piped/cloudprovider/ecs/client.go (hunk 1 above)
Comment (TaKO8Ki) [reply pkg/app/piped/cloudprovider/ecs/client.go:145]: I missed it 👍

Comment (TaKO8Ki) [reply pkg/app/piped/cloudprovider/ecs/client.go]: Done with it. We use this field mainly in case using canary deployment. So, I'm going to implement traffic routing configuration in a new PR.

File: pkg/app/piped/cloudprovider/ecs/ecs.go (hunk 2 above)
Comment (TaKO8Ki) [reply pkg/app/piped/cloudprovider/ecs/ecs.go:43]: Ok. I understand.

File: pkg/app/piped/cloudprovider/ecs/client.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/app/piped/cloudprovider/ecs/client.go]: Thanks 🙏

**Comment by khanhtc1202** [comment]
with note on https://github.com/pipe-cd/pipe/pull/1784#discussion_r603099867 📝 
/lgtm

**Comment by nghialv** [comment]
Thank you.
/approve
